server/runtime.go

```diff
@@ -73,9 +73,19 @@ func (s *Server) CreatePodSandbox(ctx context.Context, req *pb.CreatePodSandboxR
 		logDir = fmt.Sprintf("/var/log/ocid/pods/%s", name)
 	}
 
-	// TODO: construct /etc/resolv.conf based on dnsOpts.
-	dnsOpts := req.GetConfig().GetDnsOptions()
-	fmt.Println(dnsOpts)
+	dnsServers := req.GetConfig().GetDnsOptions().GetServers()
+	dnsSearches := req.GetConfig().GetDnsOptions().GetSearches()
+	resolvPath := fmt.Sprintf("%s/resolv.conf", podSandboxDir)
+	if err := parseDNSOptions(dnsServers, dnsSearches, resolvPath); err != nil {
+		if err := cleanupFile(resolvPath); err != nil {
+			return nil, err
```

Comment:
We should preserve and return the original error as well. You could combine the two using fmt.Errorf.